synapse/federation/federation_client.py

```diff
@@ -358,11 +358,33 @@ async def get_pdu(
             The requested PDU, or None if we were unable to find it.
```

Comment:
it would be good to update the docstring to say that we return a copy that is safe to modify, since that's not normally the case.